airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -712,7 +712,160 @@ public void testResetCancelsRunningSync() throws Exception {
+
+    // Set the source to a version that does not support per-stream state
+    LOGGER.info("Setting source connector to pre-per-stream state version {}...",
+        AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_LEGACY_CONNECTOR_VERSION);
+    testHarness.updateSourceDefinitionVersion(sourceDefinitionId, AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_LEGACY_CONNECTOR_VERSION);
+
+    final SyncMode syncMode = SyncMode.INCREMENTAL;
+    final DestinationSyncMode destinationSyncMode = DestinationSyncMode.APPEND;
+    catalog.getStreams().forEach(s -> s.getConfig()
+        .syncMode(syncMode)
+        .cursorField(List.of(COLUMN_ID))
+        .destinationSyncMode(destinationSyncMode));
+    final UUID connectionId =
+        testHarness.createConnection(connectionName, sourceId, destinationId, List.of(operationId), catalog, null).getConnectionId();
+    LOGGER.info("Beginning {} sync 1", testInfo.getDisplayName());
+
+    final JobInfoRead connectionSyncRead1 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead1.getJob());
+    LOGGER.info("state after sync 1: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
+
+    // get contents of source before mutating records.
+    final Database sourceDatabase = testHarness.getSourceDatabase();
+    final List<JsonNode> expectedRecords = testHarness.retrieveSourceRecords(sourceDatabase, STREAM_NAME);
+
+    // Set source to a version that supports per-stream state
+    testHarness.updateSourceDefinitionVersion(sourceDefinitionId, AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_PER_STREAM_STATE_CONNECTOR_VERSION);
+    LOGGER.info("Upgraded source connector per-stream state supported version {}.",
+        AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_PER_STREAM_STATE_CONNECTOR_VERSION);
+
+    // sync one more time. verify that nothing has been synced
+    LOGGER.info("Starting {} sync 3", testInfo.getDisplayName());
+    final JobInfoRead connectionSyncRead3 =
+        apiClient.getConnectionApi().syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead3.getJob());
+    LOGGER.info("state after sync 3: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+
+    final JobInfoRead syncJob = apiClient.getJobsApi().getJobInfo(new JobIdRequestBody().id(connectionSyncRead3.getJob().getId()));
+    final Optional<AttemptInfoRead> result = syncJob.getAttempts().stream()
+        .sorted((a, b) -> Long.compare(b.getAttempt().getEndedAt(), a.getAttempt().getEndedAt()))
+        .findFirst();
+
+    assertTrue(result.isPresent());
+    assertEquals(0, result.get().getAttempt().getRecordsSynced());
```

Comment:
Should we check that nothing has been read here (The value of what is in AttemptStats). Like if we read the data but we don't consider as sync because of it being duplicated.

Reply:
Done